### PR TITLE
[FLINK-26995][build] Enable fork-reuse for all unit tests 

### PR DESCRIPTION
**Pull request by zentol**
With this PR we reuse JVMs for all unit tests, Previously this was not the case for flink-runtime, flink-streaming-java and flink-tests.

Do that end:
- some tests were fixed to work with fork-reuse enabled
- some unit tests are now IT cases as they inherently cannot work with fork-reuse being enabled
- a few IT cases were marked as unit tests

This provides a significant boost to the test execution times, saving ~20 minutes in flink-runtime and ~5m in flink-streaming java.

### CODE REVIEW

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java [hunk 1]
```diff
@@ -228,7 +228,7 @@
 import static org.mockito.Mockito.when;
 
 /** Tests for {@link StreamTask}. */
-public class StreamTaskTest extends TestLogger {
+public class StreamTaskITCase extends TestLogger {
```
Comment (pnowojski) [inline flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java:231]: Which tests are causing trouble? I would guess only one or two? If so, can we maybe extract those offending tests to an `ITCase` file? There is a huge family of various `StreamTask` test files. Having (most of) them in the files named according to the same convention would help avoid future confusions.

**Review by pnowojski — COMMENTED** [review]
I think it mostly LGTM. I have some doubts whether we should mark unit tests as "ITCase` just because they are using some static resource - at least in some of the cases.

File: flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java (hunk 1 above)
Comment (zentol) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java:231]: It should be those that rely on `testRecordWriterClosedOnError`, so `testFailInEndOfConstructor` and the 3 `testRecordWriterClosedOnTransit*` tests.

Comment (zentol) [reply flink-streaming-java/src/test/java/org/apache/flink/streaming/runtime/tasks/StreamTaskITCase.java:231]: I will look into splitting the test.

**Comment by zentol** [comment]
> I have some doubts whether we should mark unit tests as "ITCase` just because they are using some static resource

That's a fair point. I went over the tests and split those where it was possible.
These tests remained full IT Cases:
* `FileUploadHandlerTest` checks the DeleteOnExitHooks in each test
* `StreamTaskTimerTest` checks the TRIGGER_THREAD_GROUP after each test

The renaming of TaskManagerRunnerTest was reverted, but I removed the line where we null the security manager.

**Review by pnowojski — APPROVED** [review]
Thanks for the update, I think LGTM

**Pull request merged**